dlp/inspectStringRep.js

```diff
@@ -0,0 +1,137 @@
+
+// sample-metadata:
+//  title: Inspects strings
+//  description: Inspect a string using the Data Loss Prevention API.
+//  usage: node inspectString.js my-project string minLikelihood maxFindings infoTypes customInfoTypes includeQuote
+
+function main(
+  projectId,
+  repLocation,
+  string,
+  minLikelihood,
+  maxFindings,
+  infoTypes,
+  customInfoTypes,
+  includeQuote
+) {
+  [infoTypes, customInfoTypes] = transformCLI(infoTypes, customInfoTypes);
+
+  // [START dlp_inspect_string]
+  // Imports the Google Cloud Data Loss Prevention library
+  const DLP = require('@google-cloud/dlp');
+
+  // Assemble the regional endpoint url using provided rep location
+  const opts = {
+    apiEndpoint: `dlp.${repLocation}.rep.googleapis.com`,
+  };
+
+  // Instantiates a client
+  const dlp = new DLP.DlpServiceClient(opts);
+
+  // The project ID to run the API call under
+  // const projectId = 'my-project';
+
+  // The string to inspect
+  // const string = 'My name is Gary and my email is gary@example.com';
+
+  // The minimum likelihood required before returning a match
+  // const minLikelihood = 'LIKELIHOOD_UNSPECIFIED';
+
+  // The maximum number of findings to report per request (0 = server maximum)
+  // const maxFindings = 0;
+
+  // The infoTypes of information to match
+  // const infoTypes = [{ name: 'PHONE_NUMBER' }, { name: 'EMAIL_ADDRESS' }, { name: 'CREDIT_CARD_NUMBER' }];
+
+  // The customInfoTypes of information to match
+  // const customInfoTypes = [{ infoType: { name: 'DICT_TYPE' }, dictionary: { wordList: { words: ['foo', 'bar', 'baz']}}},
+  //   { infoType: { name: 'REGEX_TYPE' }, regex: {pattern: '\\(\\d{3}\\) \\d{3}-\\d{4}'}}];
+
+  // Whether to include the matching string
+  // const includeQuote = true;
+
+  async function inspectString() {
+    // Construct item to inspect
+    const item = {value: string};
+
+    // Construct request
+    const request = {
+      parent: `projects/${projectId}/locations/${repLocation}`,
+      inspectConfig: {
+        infoTypes: infoTypes,
+        customInfoTypes: customInfoTypes,
+        minLikelihood: minLikelihood,
+        includeQuote: includeQuote,
+        limits: {
+          maxFindingsPerRequest: maxFindings,
+        },
+      },
+      item: item,
+    };
+
+    // Run request
+    const [response] = await dlp.inspectContent(request);
+    const findings = response.result.findings;
+    if (findings.length > 0) {
+      console.log('Findings:');
+      findings.forEach(finding => {
+        if (includeQuote) {
+          console.log(`\tQuote: ${finding.quote}`);
+        }
+        console.log(`\tInfo type: ${finding.infoType.name}`);
+        console.log(`\tLikelihood: ${finding.likelihood}`);
+      });
+    } else {
+      console.log('No findings.');
+    }
+  }
+  inspectString();
+  // [END dlp_inspect_string]
+}
+
+main(...process.argv.slice(2));
+process.on('unhandledRejection', err => {
+  console.error(err.message);
+  process.exitCode = 1;
+});
```

Comment:
Acknowledging that some inconsistency currently is present in our existing samples, nevertheless please remove the process handler.

It doesn't add value to the sample; an unhandled rejection will terminate the process and set the exitCode and provide a full stack trace, which is all this is doing (minus the stack trace).

If it's important to handle, you can simultaneously also address the warning that the call to main on line 53 ignores the returned promise by using [Promise.prototype.catch()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/catch)). For example:

```javascript
main(...process.argv.slice(2)).catch(err => {
  console.error(err);
  process.exitCode = 1;
});
```

Reply:
Removed